cpu/riscv_common/irq_arch.c

```diff
@@ -70,7 +70,7 @@ void riscv_irq_init(void)
 /**
  * @brief Global trap and interrupt handler
  */
-void handle_trap(uint32_t mcause)
+void __attribute((used)) handle_trap(uint32_t mcause)
```

Comment:
While at it: It should also be possible now to declare this function as static. As far as I can tell, it is only used by the `trap_entry` function which is defined in the same compilation unit.

Reply:
Added the `static` keyword.